### ISSUE TITLE
Eval model. AttributeError: 'NoneType' object has no attribute 'preprocess'

### ISSUE THREAD

**Issue by qqiuqq** [mbzuai-oryx/GeoChat#23]
I'm trying to reproduce it for scene categorization, but it's reporting an error in batch_geochat_scene.py.
Here is what the error is reported:
Traceback (most recent call last):
  File "/opt/data/private/RS-MLLM/GeoChat/./geochat/eval/batch_geochat_scene.py", line 149, in <module>
    eval_model(args)
  File "/opt/data/private/RS-MLLM/GeoChat/./geochat/eval/batch_geochat_scene.py", line 102, in eval_model
    image_tensor_batch = image_processor.preprocess(image_folder,crop_size ={'height': 504, 'width': 504},size = {'shortest_edge': 504}, return_tensors='pt')['pixel_values']
AttributeError: 'NoneType' object has no attribute 'preprocess'

and I checked image_processor and it is assigned to None in builder.py

**Comment by KjAeRsTuIsK** [comment]
Hi @qqiuqq , thank you for your interest. I have updated the repo, please update the code and try it once again. Let me know if it works now.

**Comment by Jack-bo1220** [comment]
same question

**Comment by qqiuqq** [comment]
> Hi @qqiuqq , thank you for your interest. I have updated the repo, please update the code and try it once again. Let me know if it works now.

it works！ tanks！

**Comment by ucasligang** [comment]
> I'm trying to reproduce it for the vqa task, but it's reporting the same error in batch_geochat_vqa.py.

I'm trying to reproduce it for the vqa task, but it's reporting the same error in batch_geochat_vqa.py. Do you know how to fix it？

**Comment by ucasligang** [comment]
> Hi @qqiuqq , thank you for your interest. I have updated the repo, please update the code and try it once again. Let me know if it works now.

I'm trying to reproduce it for the vqa task, but it's reporting the same error in batch_geochat_vqa.py. Do you know how to fix it？

**Comment by ucasligang** [comment]
> I'm trying to reproduce it for scene categorization, but it's reporting an error in batch_geochat_scene.py. Here is what the error is reported: Traceback (most recent call last): File "/opt/data/private/RS-MLLM/GeoChat/./geochat/eval/batch_geochat_scene.py", line 149, in eval_model(args) File "/opt/data/private/RS-MLLM/GeoChat/./geochat/eval/batch_geochat_scene.py", line 102, in eval_model image_tensor_batch = image_processor.preprocess(image_folder,crop_size ={'height': 504, 'width': 504},size = {'shortest_edge': 504}, return_tensors='pt')['pixel_values'] AttributeError: 'NoneType' object has no attribute 'preprocess'
> 
> and I checked image_processor and it is assigned to None in builder.py

I'm trying to reproduce it for the vqa task, but it's reporting the same error in batch_geochat_vqa.py. Do you know how to fix it？

**Comment by Hanzhong-Qiu** [comment]
I had this question as well. Here's how I fix it.

"AttributeError: 'NoneType' object has no attribute 'preprocess'" occurs at **line 81** "image_tensor_batch = image_processor.preprocess(image_folder,crop_size ={'height': 504, 'width': 504},size = {'shortest_edge': 504}, return_tensors='pt')['pixel_values']"

And the definition of "image_processor" comes from **line 32** "tokenizer, model, image_processor, context_len = load_pretrained_model(args.model_path, args.model_base, model_name)"

And the definition of "load_pretrained_model" has **line 42** "if 'geochat' in model_name.lower():"

So basically you need to have
geochat-7B which you downloaded from huggingface have a name with geochat in it.

I named that directory with "weights" at first which then keeps giving me that error. And I fixed it by change the name to "geochat".
